src/darray.jl

```diff
@@ -240,17 +240,24 @@ end
 # new DArray similar to an existing one
 DArray(init, d::DArray) = DArray(next_did(), init, size(d), procs(d), d.indices, d.cuts)
 
+Base.size(d::DArray) = d.dims
+chunktype(d::DArray{<:Any,<:Any,A}) where A = A
+arraykind(::Type{<:Array}) = Array
```

Comment:
@r-barnes if we add `arraykind(::Type{T}) where T >: Array = Array` the `promote_type` idea in  https://github.com/JuliaParallel/DistributedArrays.jl/issues/180#issuecomment-426090637 might just work. I won't have time until mid-October to look at that in earnest though so please go ahead and experiment.